ts/Extensions/Boost/BoostSeries.ts

```diff
@@ -604,180 +654,171 @@ function eachAsync(
-                point.destroyElements(); // #7557
-            }
-        }
+    if (
+        chart.boost &&
+        chart.boost.markerGroup === series.markerGroup
+    ) {
+        series.markerGroup = null as any;
     }
 
-    (
-        ['graph', 'area', 'tracker'] as
-        Array<('graph'|'area'|'tracker')>
-    ).forEach((prop): void => {
-        const seriesProp = series[prop];
-        if (seriesProp) {
-            series[prop] = seriesProp.destroy();
-        }
-    });
-
-    const zonesSeries = series as (BoostSeriesComposition&LineSeries);
-
-    if (zonesSeries.getZonesGraphs) {
-        const props = zonesSeries.getZonesGraphs(
-            [['graph', 'highcharts-graph']]
-        ) as Array<[keyof LineSeries]>;
-        props.forEach((prop): void => {
-            const zoneGraph: SVGElement = zonesSeries[prop[0]];
-            if (zoneGraph) {
-                (zonesSeries as any)[prop[0]] = zoneGraph.destroy();
-            }
+    if (chart.hoverPoints) {
+        chart.hoverPoints = chart.hoverPoints.filter(function (
+            point: Point
+        ): boolean {
+            return point.series === series;
         });
     }
+
+    if (chart.hoverPoint && chart.hoverPoint.series === series) {
+        chart.hoverPoint = null as any;
+    }
 }
 
 /**
- * Enter boost mode and apply boost-specific properties.
  * @private
- * @function Highcharts.Series#enterBoost
  */
-function seriesEnterBoost(
-    this: BoostSeriesComposition
+function onSeriesHide(
+    this: Series
 ): void {
-    const alteredByBoost = this.alteredByBoost =
-        [] as Array<BoostAlteredObject>;
-
-    // Save the original values, including whether it was an own
-    // property or inherited from the prototype.
-    (
-        ['allowDG', 'directTouch', 'stickyTracking'] as
-        Array<('allowDG'|'directTouch'|'stickyTracking')>
-    ).forEach((prop): void => {
-        alteredByBoost.push({
-            prop: prop,
-            val: this[prop],
-            own: Object.hasOwnProperty.call(this, prop)
-        });
-    });
-
-    this.allowDG = false;
-    this.directTouch = false;
-    this.stickyTracking = true;
+    const boost = this.boost;
 
-    // Prevent animation when zooming in on boosted series(#13421).
-    this.finishedAnimating = true;
-
-    // Hide series label if any
-    if (this.labelBySeries) {
-        this.labelBySeries = this.labelBySeries.destroy();
+    if (boost && boost.canvas && boost.target) {
+        if (boost.wgl) {
+            boost.wgl.clear();
+        }
+        if (boost.clear) {
+            boost.clear(this);
+        }
     }
 }
 
 /**
- * Exit from boost mode and restore non-boost properties.
+ * Performs the actual render if the renderer is
+ * attached to the series.
  * @private
- * @function Highcharts.Series#exitBoost
  */
-function seriesExitBoost(
-    this: BoostSeriesComposition
+function renderIfNotSeriesBoosting(
+    renderer: WGLRenderer,
+    series: Series,
```

Comment:
Apparently, the `renderIfNotSeriesBoosting` and `getPoint` function needs the `series` to be passed as an argument, so they are not in the series scope. Would it make more sense to attach these functions to the `chart.boost` object instead? It seems a bit strange when `series.boost.getPoint()` is not bound to the series. Unless it is a matter of organising - that it makes sense to put `getPoint` in the BoostSeries object because it basically deals with series stuff?

Reply:
I put `getPoint` to `Series.boost` to undergo the need, that `Core/Pointer` has to directly import it from `BoostSeries`, which bloats the core. In theory both do not need to sit on any boost. `Chart.boost` could be an alternative. 👍

Reply:
Okay, I leave it to you to see where you think it belongs.

Reply:
With `BoostChartAdditions` statements get longer, but I have found a way to make it simple to use.